File: CargoFlash.ROMS.Model/ULD/Stock/AWBStockHistory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.Serialization;

namespace CargoFlash.Cargo.Model.Stock
{
    [KnownType(typeof(AWBStockHistory))]
    public class AWBStockHistory
    {
        //AirlineSNo	OfficeSNo	AccountSNo	AirlineName	StartAWB	EndAWB	IssueDate	OfficeName	AgentName

        public int AirlineSNo { get; set; }
        public int OfficeSNo { get; set; }
        public int AccountSNo { get; set; }
        public string AirlineName { get; set; }
        public string StartAWB { get; set; }
        public string EndAWB { get; set; }
        public string IssueDate { get; set; }
        public string OfficeName { get; set; }
        public string AgentName { get; set; }
    }
}

File: CargoFlashCargoWebApps/JScript/Stock/AWBStockHistory.js
$(document).ready(function () {

    cfi.AutoComplete("AirlineSNo", "AirlineCode,AirlineName", "Airline", "AirlineCode", "AirlineName", ["AirlineCode", "AirlineName"], null, "contains");
    cfi.AutoComplete("OfficeSNo", "SNo,Name", "Office", "SNo", "Name", null, null, "contains");

    cfi.AutoComplete("AgentSNo", "SNo,Name", "Account", "SNo", "Name", null, "contains");
});



function GetReportData() {

    var AirlineSNo = $("#AirlineSNo").val();

    var OfficeSNo = $("#OfficeSNo").val();
    var AgentSNo = $("#AgentSNo").val();

    if (AirlineSNo != "") {
        $("#grid").kendoGrid({
            dataSource: {
                transport: {
                    read: {
                        url: "GetAirlineData",
                        datatype: 'json',
                        method: 'post',
                        data: { AirlineSNo: AirlineSNo, OfficeSNo: OfficeSNo = null ? "" : OfficeSNo, AgentSNo: AgentSNo = null ? "" : AgentSNo }
                    }
                }, parameterMap: function (data, operation) {
                    alert(JSON.stringify(data));
                    return JSON.stringify(data);
                },
                schema: {
                    model: {
                        id: "AirlineSNo",
                        fields: {

                            AirlineName: { type: "string" },
                            StartAWB: { type: "string" },
                            EndAWB: { type: "string" },
                            IssueDate: { type: "string" },
                            OfficeName: { type: "string" },
                            AgentName: { type: "string" },
                        }
                    }, data: "Data"
                },
                pageSize: 20,
                serverPaging: true,
                serverFiltering: true,
                serverSorting: true
            },
            height: 550,
            filterable: true,
            sortable: true,
            pageable: true,
            //detailInit: detailInit,
            //dataBound: function () {
            //    //this.expandRow(this.tbody.find("tr.k-master-row").first());
            //},
            columns: [
                //{ field: "AWBPrefix" },
                 { field: "AirlineName", title: "Airline Name" },
                { field: "StartAWB", title: "Start AWB" },
                { field: "EndAWB", title: "End AWB" },
                { field: "IssueDate", title: "Issue Date" },
                { field: "OfficeName", title: "Office Name" },
                { field: "AgentName", title: "Agent Name" }
            ]
        });
    }
}
